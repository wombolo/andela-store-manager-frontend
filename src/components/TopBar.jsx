import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";

export class TopBar extends Component {
  render() {
    const {profile} = this.props.auth.user;
    const { cartLength }= this.props;

    if (!profile){
      return'';
    }
    return (
      <div className='row'>
        <div className='col-7'>
          <h4>{profile.role.toUpperCase()}</h4>
        </div>
        <div className='col-3'>
          <div className='row'>
            <div className='col-3'>
              <Link to={'/cart'}>
                <h6>Cart <i className='cart-count'>{cartLength} </i></h6>
              </Link>
            </div>

            <div className='col-4'>
              <h5>{profile.firstname} {profile.lastname}</h5>
            </div>

            <div className='col-3'>
              <img src={`../assets/images/profile/${profile.image}`} className='top-bar-logo'/>
            </div>
          </div>
        </div>
      </div>
    )
  }
}


TopBar.propTypes = {
  auth: PropTypes.shape({}).isRequired,
};

export const mapStateToProps = state => ({
  auth: state.auth,
  cartLength: state.products.cart.length
});

export default connect(mapStateToProps)(TopBar);
