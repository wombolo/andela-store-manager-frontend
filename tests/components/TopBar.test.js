import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import { TopBar } from '../../src/components/TopBar.jsx';

let wrapper;

const props = {
  auth: {
    user: {
      profile: {
        id: 1,
        firstname: 'Akano',
        lastname: 'Peter',
        email: 'elkenzom1@gmail.com',
        role: 'admin',
        image: 'team-7.jpg',
        password: null
      },
      iat: 1551974375,
      exp: 1552060775
    },
    TopBarLength:2,
  }

};


const mockStore = configureStore([thunk]);

const store = mockStore({ ...props });


describe('TopBar Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<TopBar {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {
    beforeEach(() => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
          <TopBar {...props} />
          </MemoryRouter>
        </Provider>,
      );
    });

    afterEach(() => {
      wrapper.unmount();
    });

    it('handles no profile', () => {
      props.auth.user.profile = null;

      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <TopBar {...props} />
          </MemoryRouter>
        </Provider>,
      );

    });


  });
});
