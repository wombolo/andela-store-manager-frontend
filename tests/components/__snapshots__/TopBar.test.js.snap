// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-7"
  >
    <h4>
      User Account - 
      <span
        className="text-primary"
      >
        ADMIN
      </span>
    </h4>
  </div>
  <div
    className="col-3"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        <Link
          replace={false}
          to="/cart"
        >
          <h6>
            Cart 
            <i
              className="cart-count"
            >
               
            </i>
          </h6>
        </Link>
      </div>
      <Link
        replace={false}
        style={
          Object {
            "display": "contents",
          }
        }
        to="/profile/1"
      >
        <div
          className="col-4"
        >
          <h5>
            Akano
             
            Peter
          </h5>
        </div>
        <div
          className="col-3"
        >
          <img
            className="top-bar-logo"
            src=""
          />
        </div>
      </Link>
    </div>
  </div>
</div>
`;
